utils/database_utils/mongo_utils/queries/dao/abc.py

```diff
@@ -0,0 +1,176 @@
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from abc import ABC, abstractmethod
+
+from neon_sftp import NeonSFTPConnector
+
+from utils.database_utils import DatabaseController
+from utils.database_utils.mongo_utils import (
+    MongoQuery,
+    MongoCommands,
+    MongoFilter,
+    MongoLogicalOperators,
+)
+
+
+class MongoDocumentDAO(ABC):
+    def __init__(
+        self,
+        db_controller: DatabaseController,
+        sftp_connector: NeonSFTPConnector = None,
+    ):
+        self.db_controller = db_controller
+        self.sftp_connector = sftp_connector
+
+    @property
+    @abstractmethod
+    def document(self):
+        pass
+
+    def list_contains(
+        self,
+        key: str = "_id",
+        source_set: list = None,
+        aggregate_result: bool = True,
+        *args,
+        **kwargs
+    ) -> dict:
+        items = {}
+        contains_filter = self._build_contains_filter(key=key, lookup_set=source_set)
+        if contains_filter:
+            filters = kwargs.pop("filters", []) + [contains_filter]
+            items = self.list_items(filters=filters, *args, **kwargs)
+            if aggregate_result:
+                items = self.aggregate_items_by_key(key=key, items=items)
+        return items
+
+    def list_items(
+        self,
+        filters: list[MongoFilter] = None,
+        projection_attributes: list = None,
+        limit: int = None,
+        result_as_cursor: bool = True,
+    ) -> dict:
+        """
+        Lists items under provided document belonging to source set of provided column values
+
+        :param filters: filters to consider (optional)
+        :param projection_attributes: list of value keys to return (optional)
+        :param limit: limit number of returned attributes (optional)
+        :param result_as_cursor: to return result as cursor (defaults to True)
+        :returns results of FIND operation over the desired document according to applied filters
+        """
+        result_filters = {}
+        if limit:
+            result_filters["limit"] = limit
+        items = self._execute_query(
+            command=MongoCommands.FIND_ALL,
+            filters=filters,
+            result_filters=result_filters,
+            result_as_cursor=result_as_cursor,
+        )
+        # TODO: pymongo support projection only as aggregation API which is not yet implemented in project
+        if projection_attributes:
+            items = [
+                {k: v}
+                for item in items
+                for k, v in item.items()
+                if k in projection_attributes
+            ]
+        return items
+
+    def aggregate_items_by_key(self, key, items: list) -> dict:
+        aggregated_data = {}
```

Comment:
Please add a docstring like the above method

Reply:
please see https://github.com/NeonGeckoCom/pyklatchat/pull/79/commits/a94d596b9c878d8eef5c82d24d8546fcb4223211